staging/src/k8s.io/apiserver/pkg/util/flowcontrol/fairqueuing/queueset/types.go

```diff
@@ -95,6 +95,23 @@ type queue struct {
 	seatsInUse int
 }
 
+// queueWorkEstimate tracks the total work estimates comuted by taking
+// into account all requests in this queue.
+type queueWorkEstimate struct {
+	// InitialSeatsSum is the total number of initial seats
+	// associated with all requests in a given queue.
+	InitialSeatsSum int
+
+	// RequestedSeatsSum is the total number of requested seats
+	// associated with all requests in a given queue.
+	RequestedSeatsSum int
+
+	// AdditionalDurationSecondsSum is the accumulated additional duration
+	// in seconds for this queue, it is:
+	//  sum of all (FinalSeats*AdditionalLatency.Seconds()) for each request.
+	AdditionalDurationSecondsSum float64
+}
+
```

Comment:
I think only two things need to be tracked:
- the sum of the max-width, for the purpose of computing dR/dt
- the sum of the work, for the purpose of scoring in the shuffle-sharding

BTW, "work estimate" is not the best name for the pair of those two quantities.  Maybe they do not need to be put together in one struct.